semgrep-core/parsing/Parse_code.ml

```diff
@@ -210,24 +210,7 @@ let just_parse_with_lang lang file =
       *)
       run file [
         TreeSitter Parse_javascript_tree_sitter.parse;
-        Pfff (throw_tokens (fun file ->
-          let f () =
-            Parse_js.parse file
-          in
-          (* timeout already set in caller, then good to go *)
-          if !Flag.timeout <> 0.
-          then f ()
-          else begin
-            try
-              Common.timeout_function 5 (fun () ->
-                logger#info "running the pfff JS parser with 5s timeout";
-                f ()
-              )
-            with Timeout ->
-              logger#debug "Timeout, transforming in parse error";
-              raise Parsing.Parse_error
```

Comment:
This line is why I don't see how he possibly got a timeout. He should have had a parse error,
and I'm afraid with this patch we will have regressions for some users because a file will now take forever
because it does not parse with tree-sitter and we default to the one in pfff which is really bad at error recovery.
This timeout is a trick to go around bad error-recovery algorithm.
This timeout is an internal thing, that should not leak to the user.

Reply:
@aryx - I've added the screenshot for timeout error - https://github.com/returntocorp/semgrep/issues/2411#issuecomment-764681925